File: VB/InitNewRow_Editing/InitNewRowOnEditingBehavior.vb
Imports DevExpress.Mvvm.UI.Interactivity
Imports DevExpress.Xpf.Grid
Imports System.ComponentModel

Namespace InitNewRow_Editing

    Public Class InitNewRowOnEditingBehavior
        Inherits Behavior(Of TableView)

        Protected Overrides Sub OnAttached()
            MyBase.OnAttached()
            AddHandler AssociatedObject.FocusedRowHandleChanged, AddressOf OnFocusedRowHandleChanged
            AddHandler AssociatedObject.AddingNewRow, AddressOf OnAddingNewRow
        End Sub

        Protected Overrides Sub OnDetaching()
            RemoveHandler AssociatedObject.FocusedRowHandleChanged, AddressOf OnFocusedRowHandleChanged
            RemoveHandler AssociatedObject.AddingNewRow, AddressOf OnAddingNewRow
            MyBase.OnDetaching()
        End Sub

        Private isLocked As Boolean

        Private Sub OnFocusedRowHandleChanged(ByVal sender As Object, ByVal e As FocusedRowHandleChangedEventArgs)
            If Not isLocked AndAlso AssociatedObject.FocusedRowHandle = DataControlBase.NewItemRowHandle Then
                isLocked = True
                AssociatedObject.AddNewRow()
                isLocked = False
            End If
        End Sub

        Private Sub OnAddingNewRow(ByVal sender As Object, ByVal e As AddingNewEventArgs)
            e.NewObject = New DataItem() With {.Text = "New Row", .Number = 999997}
        End Sub
    End Class
End Namespace

File: CS/InitNewRow_Editing/InitNewRowOnEditingBehavior.cs
using DevExpress.Mvvm.UI.Interactivity;
using DevExpress.Xpf.Grid;
using System.ComponentModel;

namespace InitNewRow_Editing {
    public class InitNewRowOnEditingBehavior : Behavior<TableView> {
        protected override void OnAttached() {
            base.OnAttached();
            AssociatedObject.FocusedRowHandleChanged += OnFocusedRowHandleChanged;
            AssociatedObject.AddingNewRow += OnAddingNewRow;
        }
        protected override void OnDetaching() {
            AssociatedObject.FocusedRowHandleChanged -= OnFocusedRowHandleChanged;
            AssociatedObject.AddingNewRow -= OnAddingNewRow;
            base.OnDetaching();
        }

        bool isLocked;
        void OnFocusedRowHandleChanged(object sender, FocusedRowHandleChangedEventArgs e) {
            if(!isLocked && AssociatedObject.FocusedRowHandle == GridControl.NewItemRowHandle) {
                isLocked = true;
                AssociatedObject.AddNewRow();
                isLocked = false;
            }
        }

        void OnAddingNewRow(object sender, AddingNewEventArgs e) {
            e.NewObject = new DataItem() {
                Text = "New Row",
                Number = 999997
            };
        }
    }
}
